### PR TITLE
Move to Gradle sources

### PR DESCRIPTION
**Pull request by jaredsburrows**
- Upgrade to `1.5.0` Android Gradle plugin
- Move to normal Gradle sources

### CODE REVIEW

**Comment by jaredsburrows** [comment]
@manavo Can you take a look at this?

**Comment by manavo** [comment]
Hey @jaredsburrows! Thanks for this, and sorry for the delay!

More out of curiosity than anything, what is the benefit of this change? I'm guessing it is a best practice for dealing with things in Gradle?

**Comment by jaredsburrows** [comment]
@manavo Yes. I always would like to add Javadoc and update some method names. After this PR, I'll make a separate one. I really appreciate doorbell.io. Thanks for the hard work!

File: src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
     package="io.doorbell.android" >
-    
+
     <uses-permission android:name="android.permission.INTERNET" />
-    
+
     <uses-sdk
-        android:minSdkVersion="3"
-        android:targetSdkVersion="18" />
+        android:minSdkVersion="4"
```
Comment (manavo) [inline src/main/AndroidManifest.xml:7]: Is there a change that causes the increase from 3 to 4? I'm sure 3 is practically obsolete so far, but just checking!

**Comment by manavo** [comment]
I kind of have a feeling that the binary files (.jar, gradlew, gradle.bat) shouldn't be included?

File: src/main/AndroidManifest.xml (hunk 1 above)
Comment (jaredsburrows) [inline src/main/AndroidManifest.xml:7]: When I upgraded to `23`, the IDE suggested I moved to `4`. I kept it as low as possible.

In the future, I suggest add apachage manually so you can have full Android 23+ support.

**Comment by jaredsburrows** [comment]
@manavo You need the Gradle jar to download the all of the dependencies. Please look at the documentation here: https://docs.gradle.org/current/userguide/gradle_wrapper.html section 5.2.

**Comment by manavo** [comment]
Thanks for the explanation @jaredsburrows!

**Comment by manavo** [comment]
As a side note, I've been wanting for a while to change to a better HTTP client, like https://github.com/square/okhttp, but haven't had a chance yet. That should help with the Apache dependency too!

**Comment by manavo** [comment]
Thanks again for the PR @jaredsburrows :+1:

**Comment by jaredsburrows** [comment]
@manavo That sounds like a good idea.

**Pull request merged**